extensions/arc/deployment/src/main/java/io/quarkus/arc/deployment/ObserverValidationProcessor.java

```diff
@@ -0,0 +1,63 @@
+package io.quarkus.arc.deployment;
+
+import java.util.Collection;
+
+import javax.inject.Inject;
+
+import org.jboss.jandex.AnnotationInstance;
+import org.jboss.jandex.AnnotationValue;
+import org.jboss.jandex.DotName;
+import org.jboss.jandex.IndexView;
+import org.jboss.logging.Logger;
+
+import io.quarkus.arc.processor.BeanDeploymentValidator;
+import io.quarkus.arc.processor.DotNames;
+import io.quarkus.arc.processor.ObserverInfo;
+import io.quarkus.deployment.annotations.BuildProducer;
+import io.quarkus.deployment.annotations.BuildStep;
+import io.quarkus.deployment.builditem.ApplicationArchivesBuildItem;
+
+/**
+ * Validates observer methods from application classes.
+ * If an observer listening for {@code @Initialized(ApplicationScoped.class)} is found, it logs a warning.
+ */
+public class ObserverValidationProcessor {
+
+    private static final Logger LOGGER = Logger.getLogger(ObserverValidationProcessor.class.getName());
+
+    @Inject
+    BuildProducer<BeanDeploymentValidatorBuildItem> validators;
+
+    @BuildStep
+    public void validateApplicationObserver(ApplicationArchivesBuildItem applicationArchivesBuildItem) {
+        // an index of all application classes
+        IndexView applicationClassesIndex = applicationArchivesBuildItem.getRootArchive().getIndex();
```

Comment:
Actually, this index does not contain all the application classes but the classes from the root archive (usually `src/main/classes`). It does not contain classes from other modules in a multi-module project. But I think that we have the same issue with private injection points. So let's keep it this way for now...

Reply:
Hmm, didn't know that. If you know a better way, I am all ears.
For now, I'll at least adjust the javadoc.